### ISSUE TITLE
[JTC] set_hold_position results into not exactly reaching desired target

### ISSUE THREAD

**Issue by firesurfer** [ros-controls/ros2_controllers#759]
**Describe the bug**

In #558  (included in rolled out version 3.14.0)  `set_hold_position` was added which commands the current position as the target after sampling the last point of the trajectory (https://github.com/ros-controls/ros2_controllers/blob/1dac309979fb9392dc12feb6b46555582fed6fc6/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L346).

This results into not reaching the target exactly. (In our case a UR16e arm).
As far as I see there is no way to disable or tune this behavior

**To Reproduce**

Take some robot arm. Move rather fast from A to B. Check if the desired target has been reached within a tolerance of for example 0.001rad 

**Expected behavior**

In case the checks (e.g. arm is within goal tolerance) were successful set the last point of the trajectory as target

**Environment (please complete the following information):**
- iron
- ros2control : 3.14.0
-

**Comment by firesurfer** [comment]
@christophfroehlich  If I understand the code correctly the goal tolerances are just there for checking if the goal has been reached after sampling the last point of the trajectory. If the arm takes just a tiny bit of time to precisely reach the target this check would fail. 

I worked around this in the previous version of the JTC by implementing our own check after the action has been executed successfully, but this only works under the precondition that the JTC actually sets the last point of the trajectory as desired target.

**Comment by christophfroehlich** [comment]
this is not 100% correct, it checks for he duration of the goal_time parameter, not only once. 
this means, if you set a nonzero goal_time it should act as you desire

**Comment by firesurfer** [comment]
I did set the goal_time. Seems to work now. It was just confusing because there was a change in behavior compared to the previous version.

Nevertheless I think 

> We could add an option for using the desired states instead of the measured ones.

would be rather nice to have.
